doc/developers/contributing.rst

```diff
@@ -953,10 +953,14 @@ All logic behind estimator parameters,
 like translating string arguments into functions, should be done in ``fit``.
 
 Parameters can be passed to the ``fit`` method. Generally, these parameters
-should be restricted to variables that have the shape ``n_samples``.
-This allows for all relevant parameters to be sliced cleanly during
-cross-validation. All other parameters should be passed to
-``__init__`` or ``set_params``.
+should be restricted to variables that need to be sliced during
+cross-validation. Consequently, these variables would be either arrays with shape=[N] or
+array-like with shape=[N,D] where N is the number of samples. An example of
+this type of variable would be
+``sample_weight``. This interface allows all data-dependent
+parameters to be sliced cleanly during cross-validation.
+All other parameters should be passed to
+``__init__`` or set after construction using ``set_params``.
```

Comment:
yeah, i was mistaken about `GridSearchCV` compatibility with `set_params`, I thought it could only construct objects with `__init__` and I wasn't previously seeing how one could get `GridSearchCV` to work on params that you don't set at `__init__`.